tests/func/test_remote_dir.py

```diff
@@ -38,7 +51,55 @@ def remote():
         yield remote
 
 
-def test_isdir(remote):
+@pytest.fixture
+def remote_gs():
+    """Returns a RemoteGS connected to a bucket with the following structure:
+        bucket
+        ├── data
+        │  ├── alice
+        │  ├── alpha
+        │  └── subdir
+        │     ├── 1
+        │     ├── 2
+        │     └── 3
+        ├── empty_dir
+        ├── empty_file
+        └── foo
+    """
+    prefix = str(uuid.uuid4())
+    REMOTE_URL = "gs://dvc-test/" + prefix
+    remote = RemoteGS(None, {"url": REMOTE_URL})
+    teardowns = []
+
+    def put_object(file, content):
+        create_object_gs(remote.gs, "dvc-test", prefix + "/" + file, content)
+        teardowns.append(lambda: remote.remove(remote.path_info / file))
+
+    put_object("empty_dir/", "")
+    put_object("empty_file", "")
+    put_object("foo", "foo")
+    put_object("data/alice", "alice")
+    put_object("data/alpha", "alpha")
+    put_object("data/subdir/1", "1")
+    put_object("data/subdir/2", "2")
+    put_object("data/subdir/3", "3")
+    put_object("data/subdir/4/", "")
```

Comment:
I added this one, so that we could test nested directories, and demonstrate that it is not `walked`.

Reply:
@skshetry , what about `"data/subdir/empty_dir/"`?

Reply:
Sorry, I do not understand. I understood it as to, why am I creating a file with content `""` (empty quotes), right? Refer here: https://stackoverflow.com/a/38417397

If not, please clarify.

Reply:
@skshetry, no, it was more like giving a more meaningful name to an empty file,

I suggested changing `4/` to `empty_dir/`, if it is only a file (without the trailing slash), you can try `empty_file` or just `empty`.